### ISSUE TITLE
Attempt of modification Project template in File -> Settings -> Templating throws error

### ISSUE THREAD

**Issue by piotrzarzycki21** [prominic/Moonshine-IDE#535]
Many thanks for the tips. I will try to do that in this way. I just tried to modify project template in` File -> Settings -> Templating` but it throws me an error. I will have to look into that as well.

```
 Error #1009
: TypeError: Error #1009
: 	at actionScripts.utils::UtilsCore$/findFileWrapperAgainstFileLocation()
: 	at components.views.project::TreeView/handleProjectsChange()
: 	at flash.events::EventDispatcher/dispatchEvent()
: 	at mx.collections::ListCollectionView/dispatchEvent()
: 	at mx.collections::ListCollectionView/addItemsToView()
: 	at mx.collections::ListCollectionView/listChangeHandler()
: 	at flash.events::EventDispatcher/dispatchEvent()
: 	at mx.collections::ArrayList/internalDispatchEvent()
: 	at mx.collections::ArrayList/addItemAt()
: 	at mx.collections::ListCollectionView/addItemAt()
: 	at actionScripts.plugin.project::ProjectPlugin/handleAddProject()
: 	at flash.events::EventDispatcher/dispatchEvent()
: 	at actionScripts.plugin.templating::TemplatingPlugin/handleTemplateModify()
: 	at flash.events::EventDispatcher/dispatchEvent()
: 	at mx.core::UIComponent/dispatchEvent()
: 	at actionScripts.plugin.templating.settings.renderer::TemplateRenderer/modify()
: 	at actionScripts.plugin.templating.settings.renderer::TemplateRenderer/___TemplateRenderer_Button5_click()
:
```

_Originally posted by @piotrzarzycki21 in https://github.com/prominic/Moonshine-IDE/issues/532#issuecomment-486146520_

**Comment by rat-moonshine** [comment]
This should be fix now.

**Comment by JoelProminic** [comment]
I got this error when deleting a file or directory from the template:

    : Error #1009
    : TypeError: Error #1009
    : 	at MethodInfo-15698()
    : 	at components.views.project::TreeView/onFileDeletedOnServer()
    : 	at actionScripts.controllers::DeleteFileCommand/execute()
    : 	at actionScripts.locator::IDEController/execCommand()
    : 	at flash.events::EventDispatcher/dispatchEvent()
    : 	at MethodInfo-15685()
    : 	at flash.events::EventDispatcher/dispatchEvent()
    : 	at mx.core::UIComponent/dispatchEvent()
    : 	at mx.controls.alertClasses::AlertForm/removeAlert()
    : 	at mx.controls.alertClasses::AlertForm/clickHandler()
    :
    : Click here to Report a Bug 


Creating and editing a file worked fine.

**Comment by rat-moonshine** [comment]
The above break should have fixed now.

**Comment by piotrzarzycki21** [comment]
I just tested it and it's working now.

**Comment by JoelProminic** [comment]
I confirmed that I don't see any errors when adding, editing, or deleting files.  However, when I try to use the template to create a new project, I do not see any of my modifications.

**Comment by rat-moonshine** [comment]
Moonshine creates a new copy of template project to application storage area when modifying. But I couldn't found any relevant code that updates the existing application links to use the version from application storage (instead of regular location).

I tried to change and update the using-collection that used in project creation, but I noticed separate issues like not using the custom location in first attempt, no reset button and delete/rename button shown up against the modified template entry. 

A delete button is not suppose to be available against default templates, and template name also not permitted to change. The entry should show a reset button only. This seems requires some work, and involvement.
